### PR TITLE
[sdk] Add some utility functions to CoordinateBounds

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: #182 

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Add some utility functions to CoordinateBounds.</changelog>`.

### Summary of changes

Add some utility functions to CoordinateBounds: `getCenter`, `union`, `contains`.

### CODE REVIEW

**Review by Chaoba — APPROVED** [review]

File: sdk-base/src/main/java/com/mapbox/maps/ExtensionUtils.kt [hunk 1]
```diff
@@ -37,4 +40,59 @@ fun MapSnapshotInterface.bitmap(): Bitmap {
...
+    Point.fromLngLat(
+      min(this.southwest.longitude(), bounds.southwest.longitude()),
+      min(this.southwest.latitude(), bounds.southwest.latitude())
+    ),
+    // northeast
+    Point.fromLngLat(
+      max(this.northeast.longitude(), bounds.northeast.longitude()),
+      max(this.northeast.latitude(), bounds.northeast.latitude())
+    )
+  )
```
Comment (pengdev) [inline sdk-base/src/main/java/com/mapbox/maps/ExtensionUtils.kt:97]: can you also add tests?

Comment (kiryldz) [reply sdk-base/src/main/java/com/mapbox/maps/ExtensionUtils.kt:97]: I was hoping you will not notice it 😢 
Yep, will do

Comment (kiryldz) [reply sdk-base/src/main/java/com/mapbox/maps/ExtensionUtils.kt:97]: @pengdev - we don't even have `test` package for `sdk-base`.
Assuming current solution is a bit of workaround while we land better solution upstream - perhaps we could omit tests in that case?

Comment (pengdev) [reply sdk-base/src/main/java/com/mapbox/maps/ExtensionUtils.kt:106]: Since these functions involve some complex logic, I would suggest to add some unit tests to verify the corner cases. A few validation should be enough, especially around the border of the bounds and across the international date line. The support of wrapped and unwrapped coordinates should also be taken into consideration, and maybe state it clearly in the documentation.

**Comment by kiryldz** [comment]
Will be fixed later upstream.

**Comment by tobrun** [comment]
Capturing from upstream that I won't be feasible adding this functionality so we will continue with the approach here and add equivalent functionality to iOS

**Comment by tobrun** [comment]
@kiryldz you mentioned only supporting a part of the LatLngBounds functionality, what is missing?
Can you create a ticket on iOS side and indicate what they need to add? (they should be able to also use extension functions)

**Comment by cvb941** [comment]
What's the status on this?

**Comment by tobrun** [comment]
> What's the status on this?

We are patching and exposing the APIs upstream. This has the benefit we only need to maintain one codebase versus Android/iOS specifically. It will be coming in one of the next releases.
